### PR TITLE
feat: New host collector and analyzer for Kernel Configs

### PR DESCRIPTION
**Pull request by nvanthao**
## Description, Motivation and Context

This PR introduces new host collector+analyzer `Kernel Configs`
The collector will collect kernel configurations and the analyzer can check existence of kernel config key and value.

This will help with pre-flight checks for software like k0s and Embedded Cluster
https://docs.k0sproject.io/stable/external-runtime-deps/#linux-kernel-configuration

One thing to highlight in this change is that for this collector, all `pass` outcomes are evaluated, and the outcome will be set to `fail` if `when` condition does not match.

Sample spec

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: kernel-configs
spec:
  hostCollectors:
    - kernelConfigs: {}
  hostAnalyzers:
    - kernelConfigs:
        collectorName: "Kernel Configs Check"
        strict: true
        outcomes:
          - pass:
              when: "CONFIG_CGROUP_FREEZER=y"
              message: "The cgroup freezer is enabled"
          - pass:
              when: "CONFIG_NETFILTER_XTABLES=m"
              message: "There is netfilter xtable module"

```

## Checklist

- [X] New and existing tests pass locally with introduced changes.
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] The commit message(s) are informative and highlight any breaking changes
- [ ] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: pkg/collect/host_kernel_configs.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	output := NewResult()
+	output.SaveResult(c.BundlePath, HostKernelConfigsPath, bytes.NewBuffer(b))
+
+	return output, nil
+}
+
+func getKernelRelease() (string, error) {
+	out, err := exec.Command("uname", "-r").Output()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to determine kernel release")
```
Comment (DexterYan) [inline pkg/collect/host_kernel_configs.go]: maybe add more details like
failed to determine kernel release using 'uname -r'

**Review by DexterYan — DISMISSED** [review]

File: pkg/collect/host_kernel_configs.go (hunk 1 above)
Comment (nvanthao) [reply pkg/collect/host_kernel_configs.go]: many thanks Dexter I have udpated the message

**Review by DexterYan — DISMISSED** [review]

**Review by banjoh — CHANGES_REQUESTED** [review]
> One thing to highlight in this change is that for this collector, all pass outcomes are evaluated, and the outcome will be set to fail if when condition does not match.

This approach will confuse people. It differs from how [outcomes have been designed](https://troubleshoot.sh/docs/analyze/outcomes/) where when one of the outcomes evaluates to true, evaluation of outcomes for that analyser stops.

Other analysers follow an approach where inputs to evaluate against are passed in as separate parameters. Here is an example of how a kernel config analyser might look. Note that the list of kernel config values are passed in as `selectedKernelConfigs` input value, not defining each in an outcome.

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
spec:
  hostCollectors:
    - kernelConfigs: {}
  hostAnalyzers:
    - kernelConfigs:
        collectorName: "Kernel Configs Check"
        strict: true
        selectedKernelConfigs:
          - CONFIG_CGROUP_FREEZER=y
          - CONFIG_NETFILTER_XTABLES=m
        outcomes:
          - pass:
              when: "true"
              message: "Selected kernel configs are present"
          - fail:
              message: "{{ .ConfigsNotFound }} kernel configs are missing"
```

There are exceptions such as `velero` and `ceph` analysers where the outcomes are hardcoded in the analyser itself. This is acceptable because users do not define outcomes.

**Comment by banjoh** [comment]
The collector implementation looks good

**Comment by nvanthao** [comment]
> > One thing to highlight in this change is that for this collector, all pass outcomes are evaluated, and the outcome will be set to fail if when condition does not match.
> 
> This approach will confuse people. It differs from how [outcomes have been designed](https://troubleshoot.sh/docs/analyze/outcomes/) where when one of the outcomes evaluates to true, evaluation of outcomes for that analyser stop.
> 
> Other analysers follow an approach where inputs to evaluate against are passed in as separate parameters. Here is an example of how a kernel config analyser might look. Note that the list of kernel config values are passed in as `selectedKernelConfigs` input value, not defining each in an outcome.
> 
> ```yaml
> apiVersion: troubleshoot.sh/v1beta2
> kind: SupportBundle
> spec:
>   hostCollectors:
>     - kernelConfigs: {}
>   hostAnalyzers:
>     - kernelConfigs:
>         collectorName: "Kernel Configs Check"
>         strict: true
>         selectedKernelConfigs:
>           - CONFIG_CGROUP_FREEZER=y
>           - CONFIG_NETFILTER_XTABLES=m
>         outcomes:
>           - pass:
>               when: "true"
>               message: "Selected kernel configs are present"
>           - fail:
>               message: "{{ .ConfigsNotFound }} kernel configs are missing"
> ```
> 
> There are exceptions such as `velero` and `ceph` analysers where the outcomes are hardcoded in the analyser itself. This is acceptable because users do not define outcomes.

That's a great idea on `selectedKernelConfigs`, let me update the code + doc and update us

**Comment by nvanthao** [comment]
Hi @banjoh ,

I have made the requested changes, please let me know how it goes

Sample spec

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: kernel-configs
spec:
  hostCollectors:
    - kernelConfigs: {}
  hostAnalyzers:
    - kernelConfigs:
        collectorName: "Gerard Kernel Configs Test"
        strict: true
        selectedConfigs:
        - CONFIG_CGROUP_FREEZER=y
        - CONFIG_NETFILTER_XTABLES=m
        outcomes:
          - pass:
              message: "all kernel configs are available"
          - fail:
              message: "missing kernel config(s): {{ .ConfigsNotFound }}"
```

**Review by banjoh — APPROVED** [review]
LGTM

**Pull request merged**